pandas/_libs/src/ujson/python/objToJSON.c

```diff
@@ -456,7 +456,7 @@ static char *PyDateTimeToIso(PyDateTime_Date *obj, NPY_DATETIMEUNIT base,
 static char *PyDateTimeToIsoCallback(JSOBJ obj, JSONTypeContext *tc,
                                      size_t *len) {
 
-    if (!PyDateTime_Check(obj)) {
+    if (!PyDate_Check(obj)) {
         PyErr_SetString(PyExc_TypeError, "Expected datetime object");
```

Comment:
should the error message say "date" instead of "datetime"?  is the call to PyDateTimeToIso below safe if we have a date and not a datetime?

Reply:
Happy to change; was thinking about that myself just didn't want to go down a rabbit hole of cleaning this all up just yet

w.r.t your second question that error check currently doesn't actually raise an error, so I guess it's a no-op (though it probably should do something)

Reply:
Changed commented areas to date for now; note the second would segfault if you passed it something that wasn't a date object but I think out of scope to fix here